Std/Tactic/Ext.lean

```diff
@@ -171,17 +181,29 @@ elab_rules : tactic
 
 /--
 `ext1 pat*` is like `ext pat*` except it only applies one extensionality lemma instead
-of recursing as much as possible.
+of recursing as much as possible. Use `ext1?` for displaying used extensionality lemmas.
 -/
 macro "ext1" xs:(colGt ppSpace rintroPat)* : tactic =>
   if xs.isEmpty then `(tactic| apply_ext_lemma <;> intros)
   else `(tactic| apply_ext_lemma <;> rintro $xs*)
 
--- TODO
-/-- `ext1? pat*` is like `ext1 pat*` but gives a suggestion on what pattern to use -/
-syntax "ext1?" (colGt ppSpace rintroPat)* : tactic
-/-- `ext? pat*` is like `ext pat*` but gives a suggestion on what pattern to use -/
-syntax "ext?" (colGt ppSpace rintroPat)* (" : " num)? : tactic
+/-- `ext1? pat*` is like `ext1 pat*` but gives a suggestion on what pattern to use.
+(you might need to call `intros` between the suggested lemmas and consider to which
+goal the lemmas should be applied.) -/
+syntax (name := ext1Trace) "ext1?" (colGt ppSpace rintroPat)* : tactic
+
+/-- `ext? pat*` is like `ext pat*` but gives a suggestion on what pattern to use.
+(you might need to call `intros` between the suggested lemmas and consider to which
+goal the lemmas should be applied.) -/
```

Comment:
This sounds a bit awkward. Ideally we should produce exactly correct proof expressions.

Reply:
I haven't quite figured out how to track what `intros` does, but if you have a hint, I'll implement it.

I think `ext?` in lean3 did not create a working proof either, but might be wrong

Reply:
you don't need to, you can just say `try: intros`. Re: lean 3 not producing working proofs, I don't think that's an argument for reimplementing bugs.

Reply:
Good point, I'll give it a try!

Reply:
done.